src/container/disk/hash/disk_extendible_hash_table.cpp

```diff
@@ -4,131 +4,156 @@
+template <typename K, typename V, typename KC>
+auto DiskExtendibleHashTable<K, V, KC>::InsertToNewBucket(ExtendibleHTableDirectoryPage *directory, uint32_t bucket_idx,
+                                                          const K &key, const V &value) -> bool {
   return false;
 }
 
-/*****************************************************************************
- * MERGE
- *****************************************************************************/
-template <typename KeyType, typename ValueType, typename KeyComparator>
-void HASH_TABLE_TYPE::Merge(Transaction *transaction, const KeyType &key, const ValueType &value) {}
+template <typename K, typename V, typename KC>
+void DiskExtendibleHashTable<K, V, KC>::UpdateDirectoryMapping(ExtendibleHTableDirectoryPage *directory,
+                                                               uint32_t new_bucket_idx, page_id_t new_bucket_page_id,
+                                                               uint32_t new_local_depth, uint32_t local_depth_mask) {
+  throw NotImplementedException("DiskExtendibleHashTable is not implemented");
+}
 
 /*****************************************************************************
- * GETGLOBALDEPTH - DO NOT TOUCH
+ * REMOVE
  *****************************************************************************/
-template <typename KeyType, typename ValueType, typename KeyComparator>
-auto HASH_TABLE_TYPE::GetGlobalDepth() -> uint32_t {
-  table_latch_.RLock();
-  HashTableDirectoryPage *dir_page = FetchDirectoryPage();
-  uint32_t global_depth = dir_page->GetGlobalDepth();
-  assert(buffer_pool_manager_->UnpinPage(directory_page_id_, false));
-  table_latch_.RUnlock();
-  return global_depth;
+template <typename K, typename V, typename KC>
+auto DiskExtendibleHashTable<K, V, KC>::Remove(const K &key, Transaction *transaction) -> bool {
+  return false;
 }
 
-/*****************************************************************************
- * VERIFY INTEGRITY - DO NOT TOUCH
- *****************************************************************************/
-template <typename KeyType, typename ValueType, typename KeyComparator>
-void HASH_TABLE_TYPE::VerifyIntegrity() {
-  table_latch_.RLock();
-  HashTableDirectoryPage *dir_page = FetchDirectoryPage();
-  dir_page->VerifyIntegrity();
-  assert(buffer_pool_manager_->UnpinPage(directory_page_id_, false));
-  table_latch_.RUnlock();
+template <typename K, typename V, typename KC>
+void DiskExtendibleHashTable<K, V, KC>::VerifyIntegrity() const {
+  throw NotImplementedException("DiskExtendibleHashTable is not implemented");
 }
 
-/*****************************************************************************
- * TEMPLATE DEFINITIONS - DO NOT TOUCH
- *****************************************************************************/
-template class DiskExtendibleHashTable<int, int, IntComparator>;
+template <typename K, typename V, typename KC>
```

Comment:
it would be good if you can help put this in another file which is not part of the submission, this would help a lot for Gradescope similarity checks.